doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+
+This approach extends the definition of interpretable to include a requirement that the computed counterfactual be believably a member of the dataset. It turns out that minimizing this loss isn't enough to always get interpretable results. And in particular, the constructed counterfactual often doesn't look like a member of the target class.
+
+Similar to the previous method, this method can apply to both black and white box models. In the black-box case, there is still a performance cost from computing the numerical gradients.
+
+__TODO:__
+- Picture example of results including less interpretable ones.
+
+**Pros**
+- Provides more interpretable instances than the counterfactual instances' method.
+
+**Cons**
+- Requires access to the dataset to train the autoencoder
+- Requires setup and configuration in choosing $\gamma$ and $\beta$ and training the autoencoder
+- Requires domain knowledge when choosing what it means for a feature to be present or not
+
+#### Counterfactuals Guided by Prototypes
+
+| Model-types | Task-types     | Data-types  |
+| ----------- | -------------- | ----------- |
+| TF/Kera     | Classification | Tabular     |
+| Black-box   |                | Image       |
+|             |                | Categorical |
+
+
+- Black/white box method
+- Classification models
+- Tabular, image and categorical data types
+
+For this method, we add another term to the loss that optimizes for the distance between the counterfactual instance and close members of the target class. In doing this, we require interpretability also to mean that the generated counterfactual is believably a member target class and not just in the data distribution.
+
+With hyperparameters $c$, $\gamma$ and $\beta$, the loss is now given by:
+
+$$
+L(X'|X)= c(f_{t}(X') - p_{t})^2 + \beta L_{1}(X', X) + L_{2}(X', X)^2 + \gamma L_{2} (X', AE(X'))^2 +
+L_{2}(X', X_{proto})
+$$
+
+__TODO:__
+- Picture example of results.
+
+This method produces much more interpretable results. As well as this, because the proto term pushes the solution towards the target class, we can remove the prediction loss term and still obtain a viable counterfactual. This doesn't make much difference if we can compute the gradients directly from the model. If not, and we are using numerical gradients, then the $L_{pred}$ term is a significant bottleneck owing to repeated calls on the model to approximate the gradients. Thus, this method also applies to black-box models with a substantial performance gain on the previously mentioned approaches.
+
+**Pros**
+- Generates more interpretable instances that the CEM method
+- Blackbox version of the method doesn't require computing the numerical gradients
+- Applies to more data-types
+
+**Cons**
+- Requires access to the dataset to train the auto encoder or k-d tree
+- Requires setup and configuration in choosing $\gamma$, $\beta$ and $c$
+
+#### Counterfactuals with Reinforcement Learning
```

Comment:
I think this section could use a couple of schematics:
 - Schema of the training setup
 - Schema of the prediction/explain setup, with particular emphasis on the conditioning

@arnaudvl @RobertSamoilescu thoughts?

Reply:
I think in general adding simple schematics of fit (train) / predict (explain) where possible can help clarify different methods. Especially as some explainer methods require "typical" model training (e.g. CFRL) while others have a fit step which is quite different (e.g. for anchors or converting categorical instances to distances for counterfactuals).